src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/AdapterUtil.cs

```diff
@@ -762,29 +764,54 @@ internal static Version GetAssemblyVersion()
-                || dataSource.Contains(PBI_DATAWAREHOUSE3);
-        }
-
+        /// <summary>
+        /// Represents a collection of Azure SQL Server endpoint URLs for various regions and environments.
+        /// </summary>
+        /// <remarks>This array includes endpoint URLs for Azure SQL in global, Germany, US Government,
+        /// China, and Fabric environments. These endpoints are used to identify and interact with Azure SQL services 
+        /// in their respective regions or environments.</remarks>
         internal static readonly string[] s_azureSqlServerEndpoints = { AZURE_SQL,
                                                                         AZURE_SQL_GERMANY,
                                                                         AZURE_SQL_USGOV,
                                                                         AZURE_SQL_CHINA,
                                                                         AZURE_SQL_FABRIC };
-
+        
+        /// <summary>
+        /// Contains endpoint strings for Azure SQL Server on-demand services.
+        /// Each entry is a combination of the ONDEMAND_PREFIX and a specific Azure SQL endpoint string.
+        /// Example format: "ondemand.database.windows.net".
+        /// </summary>
+        internal static readonly string[] s_azureSqlServerOnDemandEndpoints = { ONDEMAND_PREFIX + AZURE_SQL,
+                                                                                ONDEMAND_PREFIX + AZURE_SQL_GERMANY,
+                                                                                ONDEMAND_PREFIX + AZURE_SQL_USGOV,
+                                                                                ONDEMAND_PREFIX + AZURE_SQL_CHINA,
+                                                                                ONDEMAND_PREFIX + AZURE_SQL_FABRIC };
+        /// <summary>
+        /// Represents a collection of endpoint identifiers for Azure Synapse and related services.
+        /// </summary>
+        /// <remarks>This array contains predefined endpoint strings used to identify Azure Synapse and
+        /// associated services, such as Fabric Data Warehouse and Power BI Data Warehouse.</remarks>
+        internal static readonly string[] s_azureSynapseEndpoints = { AZURE_SYNAPSE,
+                                                                      FABRIC_DATAWAREHOUSE,
+                                                                      PBI_DATAWAREHOUSE,
+                                                                      PBI_DATAWAREHOUSE2,
+                                                                      PBI_DATAWAREHOUSE3 };
+
+        internal static readonly string[] s_azureSynapseOnDemandEndpoints = s_azureSqlServerOnDemandEndpoints
+                                                                            .Concat(s_azureSynapseEndpoints)
+                                                                            .ToArray();
```

Comment:
```suggestion
        internal static readonly string[] s_azureSynapseOnDemandEndpoints = [.. s_azureSqlServerOnDemandEndpoints, .. s_azureSynapseEndpoints];
```

We recently updated LangVersion, so we can now use the spread operator rather than using LINQ.